src/plugins/discover/public/embeddable/saved_search_embeddable.tsx

```diff
@@ -82,73 +92,54 @@ export type SearchProps = Partial<DiscoverGridProps> &
   };
 
 export interface SearchEmbeddableConfig {
-  savedSearch: SavedSearch;
-  editUrl: string;
-  editPath: string;
-  indexPatterns?: DataView[];
   editable: boolean;
-  filterManager: FilterManager;
   services: DiscoverServices;
+  executeTriggerActions: UiActionsStart['executeTriggerActions'];
 }
 
 export class SavedSearchEmbeddable
   extends Embeddable<SearchInput, SearchOutput>
-  implements ISearchEmbeddable, FilterableEmbeddable
+  implements
+    ISearchEmbeddable,
+    FilterableEmbeddable,
+    ReferenceOrValueEmbeddable<SearchByValueInput, SearchByReferenceInput>
 {
-  private readonly savedSearch: SavedSearch;
-  private inspectorAdapters: Adapters;
-  private panelTitle: string = '';
-  private filtersSearchSource!: ISearchSource;
-  private subscription?: Subscription;
   public readonly type = SEARCH_EMBEDDABLE_TYPE;
-  private filterManager: FilterManager;
-  private abortController?: AbortController;
-  private services: DiscoverServices;
+  public readonly deferEmbeddableLoad = true;
 
+  private readonly services: DiscoverServices;
+  private readonly executeTriggerActions: UiActionsStart['executeTriggerActions'];
+  private readonly attributeService: SavedSearchAttributeService;
+  private readonly inspectorAdapters: Adapters;
+  private readonly subscription?: Subscription;
+
+  private abortController?: AbortController;
+  private savedSearch: SavedSearch | undefined;
+  private panelTitle: string = '';
+  private filtersSearchSource!: ISearchSource;
   private prevTimeRange?: TimeRange;
   private prevFilters?: Filter[];
   private prevQuery?: Query;
   private prevSort?: SortOrder[];
   private prevSearchSessionId?: string;
   private searchProps?: SearchProps;
-
+  private isInitialized?: boolean;
+  private isDestroyed?: boolean;
```

Comment:
QQ, why not re-using `destroyed`, which is contains the same functionality and is inherited?
https://github.com/elastic/kibana/blob/3e3419d6c914ac01c71514f9cf4f065d62a5ba2b/src/plugins/embeddable/public/lib/embeddables/embeddable.tsx#L63

And in case this is a good idea, to align naming `isInitialized` could be `initialized` here ... just for alignment, ... I personally prefer the `is...` naming

Reply:
No good reason 🙂 I used `isDestroyed` initially because the Lens embeddable I based this work on uses it's own `isDestroyed` property, but it seems like `destroyed` works just as well. Updated and switched to `initialized` as well here: https://github.com/elastic/kibana/pull/146849/commits/f9ee121190faa2291c32b3f448952b103aadad85.